### PR TITLE
Made template for React/Vue comparison + Seed 0.5.0

### PR DESCRIPTION
**Pull request by David-OConnor**
A start: https://github.com/seed-rs/seed/issues/208#issuecomment-561047975

### CODE REVIEW

**Comment by David-OConnor** [comment]
Added in some of the 0.5.0 changes. I'm not familiar with how you update some things like the docs-url versions throughout. More 0.5.0 changes probably need to be made too.

**Comment by David-OConnor** [comment]
@MartinKavik : Can you take a look to make sure I didn't dork something up before deploying? We can add to the comparison later, but I'd like to get the v0.5.0 changes out. I can't get it to build on my end due to not having the css_modules built. Could you also update the release checklist with more details? I don't think updating the version is necessary for guide content updates.

File: crate/guides/fetch.md [hunk 1]
```diff
@@ -59,15 +59,15 @@ fn view(model: &Model) -> Node<Msg> {
...
 
 #[wasm_bindgen]
 pub fn render() {
-    let app = seed::App::build(init, update, view)
-        .finish()
+    let app = seed::App::builder(update, view)
+        .after_mount(after_mount)
         .run();
 
     app.update(Msg::FetchData);
```
Comment (MartinKavik) [inline crate/guides/fetch.md]: Todo for next PRs

In all examples:
- `seed::App` replace with  `App` and make sure we have the line `use seed::{*, prelude::*}` at the top of the example (if necessary) ; And we probably should add `App` into `prelude`.
- `let app ...  app.update(... ` is anti-pattern. It should be refactored with `after_mount`.

**Comment by MartinKavik** [comment]
@David-OConnor : I've added one comment for the future, but everything looks good. 

> I can't get it to build on my end due to not having generated/css__classes built.

Could you create an issue in https://github.com/seed-rs/seed-quickstart-webpack and describe your problems? (e.g. what errors `yarn start` returns)

> Could you also update the release checklist with more details? I don't think updating the version is necessary for guide content updates.

I don't know if the checklist is really needed in this project - maybe we can delete it. 
We can sync version with Seed version but I'm not sure it makes sense. However we should update `seed-rs.org` content before the Seed version is released - we can add it as the condition to Seed's release checklist.

An idea - we can add versioning once the Seed is stable (version 1.x.x+), i.e. there will be a dropdown with Seed versions and when you choose a version, associated content will be rendered.

File: crate/guides/fetch.md (hunk 1 above)
Comment (David-OConnor) [reply crate/guides/fetch.md]: Updated this PR and the Rust quickstart with this import style.

**Comment by David-OConnor** [comment]
I didn't try anything on the JS end, which is probably the problem.

**Pull request merged**